### PR TITLE
Backend: Move `go` and `tystring` variable where they belong

### PR DESCRIPTION
**Pull request by Geod24**
Make the code a bit more idiomatic day by using an immutable array, and no forward definition.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/backend/ty.d [hunk 1]
```diff
@@ -336,7 +336,114 @@ uint tyrelax(tym_t ty) { return _tyrelax[tybasic(ty)]; }
 extern __gshared ubyte[TYMAX] tyequiv;
 
 /* Give an ascii string for a type      */
-extern (C) { extern __gshared const(char)*[TYMAX] tystring; }
+extern (C) __gshared const(char)*[TYMAX] tystring =
```
Comment (thewilsonator) [inline src/dmd/backend/ty.d:339]: does this still need to be `extern (C)`?

Comment (Geod24) [reply src/dmd/backend/ty.d:339]: I removed the last commit as the tests were failing. Will re-instate later.

**Comment by WalterBright** [comment]
The reason for the layout as it is is so the backend can be separately compiled file by file rather than needing to be lumped in as one compilation. Please remember that this backend is also used for the Digital Mars C++ compiler, and a few other tools. These sorts of refactorings create a lot of extra work for me.

**Comment by Geod24** [comment]
I don't see how this helps with this goal ? I am merely moving definition to the file they are used in. They can still be compiled individually, and they still need to be linked together at the end.

**Comment by WalterBright** [comment]
Think of ty.d as the "dot h" file and var.d as the "dot c" file. (That's also why they're "extern (C)".)

**Comment by Geod24** [comment]
But they're neither "dot c" nor "dot h". They're both "dot d". They both need to be linked in anyway (since the `ty.d` has functions). If we want they to be "dot h", they should be `.di`.

**Comment by WalterBright** [comment]
They are _used_ like dot h or dot c. The declarations go in the doth, the definitions in dotd.

**Comment by Geod24** [comment]
Where ? I was looking at [DigitalMars/dmc](https://github.com/DigitalMars/dmc) and couldn't find anything relevant ?
I really can't see how this would break separate compilation, unless you keep the code in sync _with C++ code_ ?

**Pull request closed without merging**